airflow/providers/google/cloud/triggers/bigquery_dts.py

```diff
@@ -0,0 +1,156 @@
+        return (
+            "airflow.providers.google.cloud.triggers.bigquery_dts.BigQueryDataTransferRunTrigger",
+            {
+                "project_id": self.project_id,
+                "config_id": self.config_id,
+                "run_id": self.run_id,
+                "poll_interval": self.poll_interval,
+                "gcp_conn_id": self.gcp_conn_id,
+                "delegate_to": self.delegate_to,
+                "location": self.location,
+                "impersonation_chain": self.impersonation_chain,
+            },
+        )
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:
+        """
+        Get Transfer Run status and if it one of the statuses which mean end of the job
+        then yield TriggerEvent object.
+        """
+        hook = self._get_async_hook()
+        while True:
+            try:
+                transfer_run: TransferRun = await hook.get_transfer_run(
+                    project_id=self.project_id,
+                    config_id=self.config_id,
+                    run_id=self.run_id,
+                )
+                state = transfer_run.state
+                self.log.info("Current state is %s", state)
+
+                if state == TransferState.SUCCEEDED:
+                    self.log.info("Job has completed it's work.")
+                    yield TriggerEvent(
+                        {
+                            "status": "success",
+                            "run_id": self.run_id,
+                            "message": "Job completed",
+                        }
+                    )
+                    return
+
+                elif state == TransferState.FAILED:
+                    self.log.info("Job has failed")
+                    yield TriggerEvent(
+                        {
+                            "status": "failed",
+                            "run_id": self.run_id,
+                            "message": "Job has failed",
+                        }
+                    )
+                    return
+
+                if state == TransferState.CANCELLED:
+                    self.log.info("Job has been cancelled.")
+                    yield TriggerEvent(
+                        {
+                            "status": "cancelled",
+                            "run_id": self.run_id,
+                            "message": "Job was cancelled",
+                        }
+                    )
+                    return
+
+                else:
+                    self.log.info("Job is still working...")
+                    self.log.info("Waiting for %s seconds", self.poll_interval)
+                    await asyncio.sleep(self.poll_interval)
+
+            except Exception as e:
+                yield TriggerEvent(
+                    {
+                        "status": "failed",
+                        "message": f"Trigger failed with exception: {str(e)}",
+                    }
+                )
+                return
```

Comment:
```suggestion
```

Reply:
Please check my response about returning statement after yield in Trigger class